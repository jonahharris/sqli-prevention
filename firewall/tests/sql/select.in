>>> Sada testovacich SQL dotazu: select
SELECT 1 + 1;
SELECT ALL name FROM authors;
SELECT DISTINCT name FROM authors;
SELECT DISTINCTROW name, surname FROM authors;
select HIGH_PRIORITY name from authors;
SELECT STRAIGHT_JOIN name FROM authors;
SELECT SQL_SMALL_RESULT name FROM authors;
SELECT SQL_BIG_RESULT name FROM authors;
SELECT SQL_BUFFER_RESULT name FROM authors;
SELECT SQL_CACHE name FROM authors;
SELECT SQL_NO_CACHE name FROM authors;
SELECT SQL_CALC_FOUND_ROWS name FROM authors;
SELECT ALL HIGH_PRIORITY name FROM authors;
SELECT ALL SQL_SMALL_RESULT name FROM authors;
SELECT DISTINCT SQL_NO_CACHE name FROM authors;
SELECT DISTINCT HIGH_PRIORITY SQL_SMALL_RESULT name FROM authors;
SELECT DISTINCTROW HIGH_PRIORITY SQL_CACHE name FROM authors;
SELECT DISTINCTROW HIGH_PRIORITY SQL_BIG_RESULT SQL_CACHE name FROM authors;
select * from authors;
select name from authors;
select `name` from authors;
SELECT `čas` FROM authors;
# Select: slozene identifikatory
SELECT t1.i FROM t1;
SELECT t1 . i FROM t1;
SELECT test.t1.i FROM t1;
SELECT test . t1 . i FROM t1;
SELECT t1.* FROM t1;
SELECT * from sqli_prevention.authors;
# Select: join
SELECT * FROM t1, t2, t3;
SELECT * FROM (SELECT t1, t2, t3) AS ts1;
SELECT * FROM t1 STRAIGHT_JOIN t2;
SELECT * FROM t1 STRAIGHT_JOIN t2 ON t1.a = t2.b;
SELECT * FROM t1 STRAIGHT_JOIN t2 ON (t1.a = t2.b);
SELECT * FROM a LEFT JOIN b USING (c1, c2, c3);
SELECT * FROM t1 NATURAL JOIN t2;
SELECT * FROM t1 JOIN t2 USING (j);
SELECT * FROM table1 LEFT JOIN table2 USING (id);
SELECT * FROM t1 NATURAL LEFT JOIN t2;
SELECT * FROM t1 NATURAL RIGHT JOIN t2;
SELECT * FROM a LEFT JOIN b ON a.c1 = b.c1 AND a.c2 = b.c2 AND a.c3 = b.c3;
SELECT * FROM t1 JOIN t2 JOIN t3 ON (i1 = i3);
SELECT t1.name, t2.salary FROM employee AS t1 INNER JOIN info AS t2 ON t1.name = t2.name;
SELECT * FROM t1 LEFT JOIN (t2, t3, t4) ON (t2.a=t1.a AND t3.b=t1.b AND t4.c=t1.c);
SELECT * FROM t1 LEFT JOIN (t2 CROSS JOIN t3 CROSS JOIN t4) ON (t2.a=t1.a AND t3.b=t1.b AND t4.c=t1.c);
SELECT left_tbl.* FROM left_tbl LEFT JOIN right_tbl ON left_tbl.id = right_tbl.id WHERE right_tbl.id IS NULL;
SELECT * FROM table1 LEFT JOIN table2 ON table1.id = table2.id LEFT JOIN table3 ON table2.id = table3.id;
SELECT artist.name, COUNT(work.artist_id) AS number_of_works FROM artist LEFT JOIN work ON artist.id = work.artist_id GROUP BY artist.id;
# Select: join + index
SELECT * FROM table1 USE INDEX (col1_index,col2_index) WHERE col1=1 AND col2=2 AND col3=3;
SELECT * FROM table1 IGNORE INDEX (col3_index) WHERE col1=1 AND col2=2 AND col3=3;
SELECT * FROM t1 USE INDEX (i1) IGNORE INDEX FOR ORDER BY (i2);
SELECT * FROM t1 USE INDEX (i1) IGNORE INDEX FOR ORDER BY (a);
SELECT * FROM t1 USE INDEX (i1) USE INDEX (i1,i1);
SELECT * FROM t1 USE INDEX FOR JOIN (i1) FORCE INDEX FOR JOIN (i2);
SELECT * FROM t1 USE INDEX () IGNORE INDEX (i2) USE INDEX (i1) USE INDEX (i2);
SELECT * FROM t1 USE INDEX (i1,i2) IGNORE INDEX (i2);
SELECT * FROM t1 USE INDEX (i1) IGNORE INDEX (i2) USE INDEX (i2);
# Select: where
SELECT t1.name, t2.salary FROM employee AS t1, info AS t2 WHERE t1.name = t2.name;
SELECT t1.name, t2.salary FROM employee t1, info t2 WHERE t1.name = t2.name;
SELECT col_name FROM tbl_name WHERE col_name > 0;
# Select: order by
SELECT * FROM seq ORDER BY i;
SELECT * FROM seq ORDER BY i DESC;
SELECT * FROM seq ORDER BY i ASC;
SELECT * FROM seq ORDER BY x,i;
SELECT college, region, seed FROM tournament ORDER BY region, seed;
SELECT college, region AS r, seed AS s FROM tournament ORDER BY r, s;
SELECT college, region, seed FROM tournament ORDER BY 2, 3;
# Select: group by
SELECT 12 AS a, a FROM t GROUP BY a;
SELECT a, COUNT(b) FROM test_table GROUP BY a ASC ORDER BY NULL;
SELECT a, COUNT(b) FROM test_table GROUP BY a DESC;
SELECT COUNT(col1) AS col2 FROM t GROUP BY col2 HAVING col2 = 2;
# Select: having
SELECT col_name FROM tbl_name HAVING col_name > 0;
# Select: limit
SELECT * FROM tbl LIMIT 5,10;
SELECT * FROM tbl LIMIT 95,18446744073;
SELECT * FROM tbl LIMIT 5;
select * from authors limit 4 offset 3;
# Select: alias
SELECT columna columnb FROM mytable;
