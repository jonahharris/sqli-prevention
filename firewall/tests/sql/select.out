>>> Sada testovacich SQL dotazu: select
NUMBER 1;NUMBER 1;ADD;SELECTNODATA 0 1;STMT;1;0;
NAME name;TABLE authors;SELECT 1 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 2 1 1;STMT;1;0;
NAME name;NAME surname;TABLE authors;SELECT 4 2 1;STMT;1;0;
NAME name;TABLE authors;SELECT 8 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 16 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 32 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 64 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 128 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 256 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 512 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 1024 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 9 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 33 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 514 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 42 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 268 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 332 1 1;STMT;1;0;
SELECTALL;TABLE authors;SELECT 0 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 0 1 1;STMT;1;0;
NAME name;TABLE authors;SELECT 0 1 1;STMT;1;0;
NAME ƒças;TABLE authors;SELECT 0 1 1;STMT;1;0;
# Select: slozene identifikatory
FIELDNAME t1.i;TABLE t1;SELECT 0 1 1;STMT;1;0;
FIELDNAME t1.i;TABLE t1;SELECT 0 1 1;STMT;1;0;
FIELDNAME test.t1.i;TABLE t1;SELECT 0 1 1;STMT;1;0;
FIELDNAME test.t1.i;TABLE t1;SELECT 0 1 1;STMT;1;0;
FIELDNAME t1.*;TABLE t1;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE sqli_prevention.authors;SELECT 0 1 1;STMT;1;0;
# Select: join
SELECTALL;TABLE t1;TABLE t2;TABLE t3;SELECT 0 1 3;STMT;1;0;
SELECTALL;NAME t1;NAME t2;NAME t3;SELECTNODATA 0 3;SUBQUERY;SUBQUERYAS ts1;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;JOIN 200;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;FIELDNAME t1.a;FIELDNAME t2.b;CMP 4;JOIN 200;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;FIELDNAME t1.a;FIELDNAME t2.b;CMP 4;JOIN 200;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE a;TABLE b;COLUMN c1;COLUMN c2;COLUMN c3;USING 3;JOIN 301;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;JOIN 400;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;COLUMN j;USING 1;JOIN 100;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE table1;TABLE table2;COLUMN id;USING 1;JOIN 301;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;JOIN 401;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;JOIN 402;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE a;TABLE b;FIELDNAME a.c1;FIELDNAME b.c1;CMP 4;FIELDNAME a.c2;FIELDNAME b.c2;CMP 4;AND;FIELDNAME a.c3;FIELDNAME b.c3;CMP 4;AND;ONEXPR;JOIN 301;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;JOIN 100;TABLE t3;NAME i1;NAME i3;CMP 4;ONEXPR;JOIN 100;SELECT 0 1 1;STMT;1;0;
FIELDNAME t1.name;FIELDNAME t2.salary;ALIAS t1;TABLE employee;ALIAS t2;TABLE info;FIELDNAME t1.name;FIELDNAME t2.name;CMP 4;ONEXPR;JOIN 101;SELECT 0 2 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;TABLE t3;TABLE t4;TABLEREFERENCES 3;FIELDNAME t2.a;FIELDNAME t1.a;CMP 4;FIELDNAME t3.b;FIELDNAME t1.b;CMP 4;AND;FIELDNAME t4.c;FIELDNAME t1.c;CMP 4;AND;ONEXPR;JOIN 301;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE t1;TABLE t2;TABLE t3;JOIN 102;TABLE t4;JOIN 102;TABLEREFERENCES 1;FIELDNAME t2.a;FIELDNAME t1.a;CMP 4;FIELDNAME t3.b;FIELDNAME t1.b;CMP 4;AND;FIELDNAME t4.c;FIELDNAME t1.c;CMP 4;AND;ONEXPR;JOIN 301;SELECT 0 1 1;STMT;1;0;
FIELDNAME left_tbl.*;TABLE left_tbl;TABLE right_tbl;FIELDNAME left_tbl.id;FIELDNAME right_tbl.id;CMP 4;ONEXPR;JOIN 301;FIELDNAME right_tbl.id;ISNULL;WHERE;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE table1;TABLE table2;FIELDNAME table1.id;FIELDNAME table2.id;CMP 4;ONEXPR;JOIN 301;TABLE table3;FIELDNAME table2.id;FIELDNAME table3.id;CMP 4;ONEXPR;JOIN 301;SELECT 0 1 1;STMT;1;0;
FIELDNAME artist.name;FIELDNAME work.artist_id; CALL 1 COUNT;ALIAS number_of_works;TABLE artist;TABLE work;FIELDNAME artist.id;FIELDNAME work.artist_id;CMP 4;ONEXPR;JOIN 301;FIELDNAME artist.id;GROUPBY 0;GROUPBYLIST 1 0;SELECT 0 2 1;STMT;1;0;
# Select: join + index
SELECTALL;INDEX col1_index;INDEX col2_index;INDEXHINT USE 2;TABLE table1;NAME col1;NUMBER 1;CMP 4;NAME col2;NUMBER 2;CMP 4;AND;NAME col3;NUMBER 3;CMP 4;AND;WHERE;SELECT 0 1 1;STMT;1;0;
SELECTALL;INDEX col3_index;INDEXHINT IGNORE 1;TABLE table1;NAME col1;NUMBER 1;CMP 4;NAME col2;NUMBER 2;CMP 4;AND;NAME col3;NUMBER 3;CMP 4;AND;WHERE;SELECT 0 1 1;STMT;1;0;
SELECTALL;INDEX i1;INDEXHINT USE 1;FOR ORDER BY;INDEX i2;INDEXHINT IGNORE 1;TABLE t1;SELECT 0 1 1;STMT;1;0;
SELECTALL;INDEX i1;INDEXHINT USE 1;FOR ORDER BY;INDEX a;INDEXHINT IGNORE 1;TABLE t1;SELECT 0 1 1;STMT;1;0;
SELECTALL;INDEX i1;INDEXHINT USE 1;INDEX i1;INDEX i1;INDEXHINT USE 2;TABLE t1;SELECT 0 1 1;STMT;1;0;
SELECTALL;FOR JOIN;INDEX i1;INDEXHINT USE 1;FOR JOIN;INDEX i2;INDEXHINT FORCE 1;TABLE t1;SELECT 0 1 1;STMT;1;0;
SELECTALL;INDEXHINT USE EMPTY;INDEX i2;INDEXHINT IGNORE 1;INDEX i1;INDEXHINT USE 1;INDEX i2;INDEXHINT USE 1;TABLE t1;SELECT 0 1 1;STMT;1;0;
SELECTALL;INDEX i1;INDEX i2;INDEXHINT USE 2;INDEX i2;INDEXHINT IGNORE 1;TABLE t1;SELECT 0 1 1;STMT;1;0;
SELECTALL;INDEX i1;INDEXHINT USE 1;INDEX i2;INDEXHINT IGNORE 1;INDEX i2;INDEXHINT USE 1;TABLE t1;SELECT 0 1 1;STMT;1;0;
# Select: where
FIELDNAME t1.name;FIELDNAME t2.salary;ALIAS t1;TABLE employee;ALIAS t2;TABLE info;FIELDNAME t1.name;FIELDNAME t2.name;CMP 4;WHERE;SELECT 0 2 2;STMT;1;0;
FIELDNAME t1.name;FIELDNAME t2.salary;ALIAS t1;TABLE employee;ALIAS t2;TABLE info;FIELDNAME t1.name;FIELDNAME t2.name;CMP 4;WHERE;SELECT 0 2 2;STMT;1;0;
NAME col_name;TABLE tbl_name;NAME col_name;NUMBER 0;CMP 2;WHERE;SELECT 0 1 1;STMT;1;0;
# Select: order by
SELECTALL;TABLE seq;NAME i;ORDERBY 0;ORDERBYLIST 1;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE seq;NAME i;ORDERBY 2;ORDERBYLIST 1;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE seq;NAME i;ORDERBY 1;ORDERBYLIST 1;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE seq;NAME x;ORDERBY 0;NAME i;ORDERBY 0;ORDERBYLIST 2;SELECT 0 1 1;STMT;1;0;
NAME college;NAME region;NAME seed;TABLE tournament;NAME region;ORDERBY 0;NAME seed;ORDERBY 0;ORDERBYLIST 2;SELECT 0 3 1;STMT;1;0;
NAME college;NAME region;ALIAS r;NAME seed;ALIAS s;TABLE tournament;NAME r;ORDERBY 0;NAME s;ORDERBY 0;ORDERBYLIST 2;SELECT 0 3 1;STMT;1;0;
NAME college;NAME region;NAME seed;TABLE tournament;NUMBER 2;ORDERBY 0;NUMBER 3;ORDERBY 0;ORDERBYLIST 2;SELECT 0 3 1;STMT;1;0;
# Select: group by
NUMBER 12;ALIAS a;NAME a;TABLE t;NAME a;GROUPBY 0;GROUPBYLIST 1 0;SELECT 0 2 1;STMT;1;0;
NAME a;NAME b; CALL 1 COUNT;TABLE test_table;NAME a;GROUPBY 1;GROUPBYLIST 1 0;NULL;ORDERBY 0;ORDERBYLIST 1;SELECT 0 2 1;STMT;1;0;
NAME a;NAME b; CALL 1 COUNT;TABLE test_table;NAME a;GROUPBY 2;GROUPBYLIST 1 0;SELECT 0 2 1;STMT;1;0;
NAME col1; CALL 1 COUNT;ALIAS col2;TABLE t;NAME col2;GROUPBY 0;GROUPBYLIST 1 0;NAME col2;NUMBER 2;CMP 4;HAVING;SELECT 0 1 1;STMT;1;0;
# Select: having
NAME col_name;TABLE tbl_name;NAME col_name;NUMBER 0;CMP 2;HAVING;SELECT 0 1 1;STMT;1;0;
# Select: limit
SELECTALL;TABLE tbl;NUMBER 5;NUMBER 10;LIMIT 2;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE tbl;NUMBER 95;NUMBER 1266874889;LIMIT 2;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE tbl;NUMBER 5;LIMIT 1;SELECT 0 1 1;STMT;1;0;
SELECTALL;TABLE authors;NUMBER 4;NUMBER 3;LIMIT 3;SELECT 0 1 1;STMT;1;0;
# Selec: alias
NAME columna;ALIAS columnb;TABLE mytable;SELECT 0 1 1;STMT;1;0;
